### PR TITLE
Update building-and-testing-nodejs examples

### LINKED ISSUE #25313: Update node version examples from 10 - 16x to 14 - 20x
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs

### What changes are you suggesting?

The node versions used in this build guide are severely out of date and no longer supported. For users creating a new project, they'd be much better off with a more recent default example set to base off.

https://github.com/github/docs/pull/25312

### Additional information

https://nodejs.dev/en/about/releases/

I used this as a guide for which versions to use as examples when making my update recs PR.

### PR DESCRIPTION
**Pull request by Rylab**
### Why: The docs recommend using extremely deprecated versions of node. Updated to newer stable versions instead.

Closes: https://github.com/github/docs/issues/25313

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by skedwards88 — COMMENTED** [review]
Thanks for updating this! I just have one question @Rylab : You mention that a workflow failed with Node version 14 (https://github.com/github/docs/issues/25313#issuecomment-1530541751). Should these examples exclude version 14?

**Comment by Rylab** [comment]
> Thanks for updating this! I just have one question @Rylab : You mention that a workflow failed with Node version 14 ([#25313 (comment)](https://github.com/github/docs/issues/25313#issuecomment-1530541751)). Should these examples exclude version 14?

16 is the minimum node version most common frameworks recommend, so yeah going a step further and removing 14 from the examples is probably a good idea, too.

File: content/actions/automating-builds-and-tests/building-and-testing-nodejs.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x, 14.x, 15.x]
+        node-version: [14.x, 16.x, 18.x, 20.x]
```
Comment (skedwards88) [inline content/actions/automating-builds-and-tests/building-and-testing-nodejs.md:59]: Also removing version 14 based on https://github.com/github/docs/pull/25312#issuecomment-1532802449
```suggestion
        node-version: [16.x, 18.x, 20.x]
```

File: content/actions/automating-builds-and-tests/building-and-testing-nodejs.md [hunk 2]
```diff
@@ -77,14 +77,14 @@ The easiest way to specify a Node.js version is by using the `setup-node` action
...
 
-The starter workflow includes a matrix strategy that builds and tests your code with four Node.js versions: 10.x, 12.x, 14.x, and 15.x. The 'x' is a wildcard character that matches the latest minor and patch release available for a version. Each version of Node.js specified in the `node-version` array creates a job that runs the same steps.
+The starter workflow includes a matrix strategy that builds and tests your code with four Node.js versions: 14.x, 16.x, 18.x, and 20.x. The 'x' is a wildcard character that matches the latest minor and patch release available for a version. Each version of Node.js specified in the `node-version` array creates a job that runs the same steps.
```
Comment (skedwards88) [inline content/actions/automating-builds-and-tests/building-and-testing-nodejs.md:80]:
```suggestion
The starter workflow includes a matrix strategy that builds and tests your code with three Node.js versions: 16.x, 18.x, and 20.x. The 'x' is a wildcard character that matches the latest minor and patch release available for a version. Each version of Node.js specified in the `node-version` array creates a job that runs the same steps.
```

File: content/actions/automating-builds-and-tests/building-and-testing-nodejs.md [hunk 3]
```diff
@@ -77,14 +77,14 @@ The easiest way to specify a Node.js version is by using the `setup-node` action
...
 
 Each job can access the value defined in the matrix `node-version` array using the `matrix` context. The `setup-node` action uses the context as the `node-version` input. The `setup-node` action configures each job with a different Node.js version before building and testing code. For more information about matrix strategies and contexts, see "[AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix)" and "[AUTOTITLE](/actions/learn-github-actions/contexts)."
 
 ```yaml{:copy}
 strategy:
   matrix:
-    node-version: [10.x, 12.x, 14.x, 15.x]
+    node-version: [14.x, 16.x, 18.x, 20.x]
```
Comment (skedwards88) [inline content/actions/automating-builds-and-tests/building-and-testing-nodejs.md:87]:
```suggestion
    node-version: [16.x, 18.x, 20.x]
```

File: content/actions/automating-builds-and-tests/building-and-testing-nodejs.md [hunk 4]
```diff
@@ -99,7 +99,7 @@ Alternatively, you can build and test with exact Node.js versions.
 ```yaml{:copy}
 strategy:
   matrix:
-    node-version: [8.16.2, 10.17.0]
+    node-version: [10.17.0, 17.9.0]
```
Comment (skedwards88) [inline content/actions/automating-builds-and-tests/building-and-testing-nodejs.md:102]:
```suggestion
    node-version: [17.9.0, 18.12.0]
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for updating this! I also removed the remaining versions <=14 based on https://github.com/github/docs/pull/25312#issuecomment-1532802449 I'll incorporate those changes then get this merged for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**